modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/AttachmentUtil.java

```diff
@@ -259,13 +264,128 @@ public static CPAttachmentFileEntry addOrUpdateCPAttachmentFileEntry(
 			GetterUtil.get(attachmentUrl.getGalleryEnabled(), true),
 			getTitleMap(null, attachmentUrl.getTitle()),
 			_getJSON(
-				cpDefinitionOptionRelService, cpDefinitionOptionValueRelService,
-				cpOptionService, attachmentUrl.getOptions(), classPK,
+				null, cpDefinitionOptionRelService,
+				cpDefinitionOptionValueRelService, cpOptionService,
+				attachmentUrl.getOptions(), classPK,
 				serviceContext.getCompanyId()),
 			GetterUtil.getDouble(attachmentUrl.getPriority()), type,
 			serviceContext);
 	}
 
+	public static CPAttachmentFileEntry addOrUpdateCPAttachmentFileEntry(
+			long groupId,
+			CPAttachmentFileEntryService cpAttachmentFileEntryService,
+			CPDefinitionOptionRelService cpDefinitionOptionRelService,
+			CPDefinitionOptionValueRelService cpDefinitionOptionValueRelService,
+			CPOptionService cpOptionService, DLAppService dlAppService,
+			ModelResourcePermission<DLFileEntry>
+				dlFileEntryModelResourcePermission,
+			GroupLocalService groupLocalService,
+			UniqueFileNameProvider uniqueFileNameProvider,
+			Attachment attachment, long classNameId, long classPK, int type,
+			ServiceContext serviceContext)
+		throws Exception {
+
+		ServiceContext dlFileEntryCloneServiceContext =
+			(ServiceContext)serviceContext.clone();
+
+		dlFileEntryCloneServiceContext.setExpandoBridgeAttributes(
+			new LinkedHashMap<>());
+
+		dlFileEntryCloneServiceContext.setWorkflowAction(
+			WorkflowConstants.ACTION_PUBLISH);
+
+		long fileEntryId = GetterUtil.getLong(attachment.getFileEntryId());
+
+		if (fileEntryId == 0) {
+			FileEntry fileEntry;
+
+			String attachmentFileEntryGroupExternalReferenceCode =
+				GetterUtil.getString(
+					attachment.getFileEntryGroupExternalReferenceCode());
+
+			if (attachmentFileEntryGroupExternalReferenceCode.isEmpty()) {
+				fileEntry = addFileEntry(
+					attachment, uniqueFileNameProvider,
+					dlFileEntryCloneServiceContext);
+			}
+			else {
+				Group group =
+					groupLocalService.fetchGroupByExternalReferenceCode(
+						attachmentFileEntryGroupExternalReferenceCode,
+						serviceContext.getCompanyId());
+
+				if (group != null) {
+					fileEntry =
+						dlAppService.getFileEntryByExternalReferenceCode(
+							attachment.getFileEntryExternalReferenceCode(),
+							group.getGroupId());
+				}
+				else {
+					throw new NoSuchGroupException();
+				}
+			}
+
+			if (fileEntry != null) {
+				fileEntryId = fileEntry.getFileEntryId();
+			}
+		}
+		else {
+			dlFileEntryModelResourcePermission.check(
+				PermissionThreadLocal.getPermissionChecker(), fileEntryId,
+				ActionKeys.VIEW);
+		}
+
```

Comment:
please simplify, there are a lots of examples of how to do this piece of code.
First check the erc, then the id and finally fall back to the fileentry